### ISSUE TITLE
Cannot select test from parent spec

### ISSUE THREAD

**Issue by ldaley** [spockframework/spock#1235]
# Issue description

Tests inherited from super classes cannot be selected/filtered with Spock 2 (m3), where they could be with Spock 1.

# How to reproduce

[spock-2-superclass-filter-bug.zip](https://github.com/spockframework/spock/files/5535185/spock-2-superclass-filter-bug.zip)

Attached is a complete Groovy project, that can use Spock1 or 2. 

Works:

```
./gradlew test --tests *ChildSpec.parent -Pspock1
```

Fails (uses Spock2):

```
./gradlew test --tests *ChildSpec.parent
```

**Comment by leonard84** [comment]
@marcphilipp could you shed some light on how selection works in gradle and what might be the issue?

**Comment by marcphilipp** [comment]
The culprit is probably this filter in Gradle:
https://github.com/gradle/gradle/blob/46d60a4c6d72ce8880882e76a73d8db25d8aed9a/subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java#L171-L194

**Comment by leonard84** [comment]
I haven't debugged it yet but my guess is that it is probably that line here 
`return matcher.matchesTest(methodSource.getClassName(), methodSource.getMethodName())` 
https://github.com/gradle/gradle/blob/46d60a4c6d72ce8880882e76a73d8db25d8aed9a/subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java#L179

I think that the `methodSource.getClassName()` will return the declaring class and not the inheriting class and thus be filtered out.

**Comment by marcphilipp** [comment]
The `MethodSource` is supposed to contain the declaring class, hence "source", and was introduced to allow IDE navigation. The filtering code could check for the parent descriptor's `ClassSource` in this case instead of using the `MethodSource`'s class name.

**Comment by leonard84** [comment]
@ldaley it was fixed in gradle, so I'm closing this issue now.

**Comment by kriegaex** [comment]
Shouldn't this stay open in order to track it upstream and then document here and in the next release notes in which Gradle release it is fixed? It would be nice to be able to tell our users something like "use Gradle x.y.z and you will be fine".

**Comment by leonard84** [comment]
I don't really want to manage external tool release stuff, I mean the issues are linked as far as I can see it, it should be in 7.0 RC1
If you are offering to track these kinds of 3rd party issues in the future, then we can keep them open.

**Comment by kriegaex** [comment]
I suggest the @ldaley tracks it, re-tests with the future Gradle release containing the fix and reports back. He is the main stakeholder here.

**Comment by marcphilipp** [comment]
I think it's fine to close the issue since the fix has been merged into Gradle and will be part of 7.0. Whoever will search for a similar issue should still find this one.

**Comment by ldaley** [comment]
+1 to @marcphilipp's comment.
